lib/db/lowlevel.go

```diff
@@ -9,9 +9,23 @@ package db
 import (
 	"bytes"
 	"encoding/binary"
+	"time"
 
 	"github.com/syncthing/syncthing/lib/db/backend"
 	"github.com/syncthing/syncthing/lib/protocol"
+	"github.com/syncthing/syncthing/lib/sync"
+	"github.com/willf/bloom"
+)
+
+const (
+	// We set the bloom filter capacity to handle 100k individual block lists
+	// with a false positive probability of 1% for the first pass. Once we know
+	// how many block lists we have we will use that number instead, if it's
+	// more than 100k. For fewer than 100k block lists we will just get better
+	// false positive rate instead.
+	blockGCBloomCapacity          = 100000
+	blockGCBloomFalsePositiveRate = 0.01 // 1%
+	blockGCInterval               = 13 * time.Hour
```

Comment:
The GC thing is a full walk of the database, so can take potentially minutes during which a lock is held. It's not important that it happens immediately so I don't want to block startup (or even sync stuff that would otherwise happen at startup) for it. I picked 13 hours as something that felt often enough yet hopefully not impacting and doesn't divide 24 hours so that it should slide to different times of day. That avoids it always happening during peak hours for example, if you have peak hours.

If we were to make this configurable I think it should be an env var as it's not even interesting enough to be an advanced config for almost all users, I think...